skeleton/default.nix

```diff
@@ -0,0 +1,60 @@
+{ system ? builtins.currentSystem
+, crossSystem ? null
+, config ? {}
+# Import IOHK common nix lib
+, iohkLib ? import ./nix/iohk-common.nix { inherit system crossSystem config; }
+# Use nixpkgs pin from iohkLib
+, pkgs ? iohkLib.pkgs
+}:
+
+with import ./nix/util.nix { inherit pkgs; };
+
+let
+  haskell = iohkLib.nix-tools.haskell { inherit pkgs; };
+  src = iohkLib.cleanSourceHaskell ./.;
+
+  # Example of using a package from iohk-nix
+  # TODO: Declare packages required by the build.
+  inherit (iohkLib.rust-packages.pkgs) jormungandr;
+
+  # Import the Haskell package set.
+  haskellPackages = import ./nix/default.nix {
+    inherit pkgs haskell src;
+    # Pass in any extra programs necessary for the build as function arguments.
+    # TODO: Declare packages required by the build
+    inherit jormungandr;
+    inherit (pkgs) cowsay;
+    # Provide cross-compiling secret sauce
+    inherit (iohkLib.nix-tools) iohk-extras iohk-module;
+  };
+
+in {
+  inherit pkgs iohkLib src haskellPackages;
+  inherit (haskellPackages.iohk-skeleton.identifier) version;
+
+  # Grab the executable component of our package.
+  inherit (haskellPackages.iohk-skeleton.components.exes)
+    iohk-skeleton;
+
+  tests = collectComponents "tests" isIohkSkeleton haskellPackages;
+  benchmarks = collectComponents "benchmarks" isIohkSkeleton haskellPackages;
+
+  # This provides a development environment that can be used with nix-shell or
+  # lorri. See https://input-output-hk.github.io/haskell.nix/user-guide/development/
+  shell = haskellPackages.shellFor {
+    name = "iohk-skeleton-shell";
+    # TODO: List all local packages in the project.
+    packages = ps: with ps; [
+      iohk-skeleton
+    ];
+    # These programs will be available inside the nix-shell.
+    buildInputs =
+      with pkgs.haskellPackages; [ hlint stylish-haskell weeder ghcid lentil ]
+      # TODO: Add your own packages to the shell.
+      ++ [ jormungandr ];
+  };
+
+  checks.lint-fuzz = pkgs.callPackage ./nix/check-lint-fuzz.nix {};
+
+  docs = import ./docs/default.nix { inherit pkgs; };
```

Comment:
Why not `callPackage`? I know it probably doesn't matter here, but as a naive reader I would wonder.

Reply:
Yeah, I'd recommend callPackage and not pass all of pkgs :)